spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/configdata/ConfigDataProperties.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config.configdata;
+
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.context.config.ConfigDataLocationResolverContext;
+import org.springframework.cloud.kubernetes.commons.KubernetesClientProperties;
+import org.springframework.cloud.kubernetes.commons.config.ConfigMapConfigProperties;
+import org.springframework.cloud.kubernetes.commons.config.SecretsConfigProperties;
+
+import static org.springframework.cloud.kubernetes.commons.config.ConfigUtils.registerSingle;
+
+public record ConfigDataProperties(KubernetesClientProperties clientProperties,
+		ConfigMapConfigProperties configMapProperties, SecretsConfigProperties secretsProperties) {
+
+	static Registrar of(ConfigDataLocationResolverContext context) {
```

Comment:
the idea here is once you call `of`, the only option you have after is `register` (this is partial functions kind of), which will register the properties and give them back in form of the `ConfigDataProperties` instance. My sole purpose of the PR was around this thing, mainly, to decouple and make configdata slightly easier to reason about.